Thus far, failures seem real.
```
_______________________ FLAKE8-check(ignoring E501 W504) _______________________
/home/travis/build/astropy/astropy-bot/astropy_bot/tests/test_autolabel.py:63:1: W391 blank line at end of file
/home/travis/build/astropy/astropy-bot/astropy_bot/tests/test_autolabel.py:63:1: W293 blank line contains whitespace
___________________ TestAutolabel.test_autolabel_subpackages ___________________
self = <astropy_bot.tests.test_autolabel.TestAutolabel object at 0x7fc5b18d5cf8>
app = <Flask 'testbot'>
    def test_autolabel_subpackages(self, app):
        with app.app_context():
            autolabel_subpackages(self.pr_handler, self.repo_handler)
    
        # make sure 'Docs' stays in there:
        assert 'Docs' in self.pr_handler.labels
    
        # now check that all the subpackage labels are added:
>       assert 'io.fits' in self.pr_handler.labels
E       AssertionError: assert 'io.fits' in ['Docs']
E        +  where ['Docs'] = <MagicMock id='140487064116752'>.labels
E        +    where <MagicMock id='140487064116752'> = <astropy_bot.tests.test_autolabel.TestAutolabel object at 0x7fc5b18d5cf8>.pr_handler
```

--- a/astropy_bot/autolabel.py
+++ b/astropy_bot/autolabel.py
@@ -13,7 +13,7 @@ def get_subpackage_labels(files, all_labels):
 
             if subpkg:
                 for i in range(len(subpkg)):
-                    dot_name = '.'.join(subpkg[:i+1])
+                    dot_name = '.'.join(subpkg[:i + 1])
                     if dot_name in all_labels:
                         labels.add(dot_name)
 

--- a/astropy_bot/autolabel.py
+++ b/astropy_bot/autolabel.py
@@ -2,23 +2,40 @@ from os import path, sep
 from baldrick.plugins.github_pull_requests import pull_request_handler
 
 
-@pull_request_handler
-def autolabel_subpackages(pr_handler, repo_handler):
-    files = pr_handler.get_modified_files()
-    all_labels = repo_handler.get_all_labels()
-    existing_labels = set(pr_handler.labels)
+def get_subpackage_labels(files, all_labels):
+    labels = set()
 
-    for file in files:
-        subdir = path.dirname(file)
+    for file_ in files:
+        subdir = path.dirname(file_)
 
         if subdir:
             root, *subpkg = subdir.split(sep)
 
             if subpkg:
-                dot_name = '.'.join(subpkg)
-                if dot_name in all_labels and dot_name not in existing_labels:
-                    existing_labels.add(dot_name)
+                for i in range(len(subpkg)):
+                    dot_name = '.'.join(subpkg[:i+1])
+                    if dot_name in all_labels:
+                        labels.add(dot_name)
 
-    pr_handler.set_labels(existing_labels)
+    return labels
+
+
+@pull_request_handler
+def autolabel(pr_handler, repo_handler):
+
+    al_config = repo_handler.get_config_value("autolabel", {})
+    files = pr_handler.get_modified_files()
+    all_labels = repo_handler.get_all_labels()
+    pr_labels = set(pr_handler.labels)
+    new_labels = set()
+
+    if al_config.get('subpackages', False):
+        labels = get_subpackage_labels(files, all_labels)
+        new_labels = new_labels.union(labels)
+
+    # TODO: add other auto-labeling logic here
+
+    if new_labels:
+        pr_handler.set_labels(pr_labels.union(new_labels))
 
     return None

--- a/astropy_bot/tests/test_autolabel.py
+++ b/astropy_bot/tests/test_autolabel.py
@@ -11,14 +11,12 @@ class TestAutolabel:
 
     def setup_method(self, method):
 
-        self.files = [
-            {'filename': 'CHANGES.rst'},
-            {'filename': 'astropy/io/fits/convenience.py'},
-            {'filename': 'astropy/io/fits/fitstime.py'},
-            {'filename': 'astropy/io/fits/tests/test_fitstime.py'},
-            {'filename': 'astropy/coordinates/baseframe.py'},
-            {'filename': 'astropy/visualization/wcsaxes/coordinate_helpers.py'}
-        ]
+        self.files = ['CHANGES.rst', 'astropy/io/fits/convenience.py',
+                      'astropy/io/fits/fitstime.py',
+                      'astropy/io/fits/tests/test_fitstime.py',
+                      'astropy/coordinates/baseframe.py',
+                      'astropy/visualization/wcsaxes/coordinate_helpers.py']
+
         self.all_labels = ['analytic_functions', 'config', 'constants',
                            'convolution', 'coordinates', 'cosmology',
                            'io.ascii', 'io.fits', 'io.misc', 'io.misc.asdf',
@@ -33,9 +31,9 @@ class TestAutolabel:
         self.pr_handler.labels = ['Docs']
         self.pr_handler.milestone = None
         self.pr_handler.set_labels = self.set_labels
+        self.pr_handler.get_modified_files = self.get_modified_files
 
         self.repo_handler = MagicMock()
-        self.repo_handler.get_modified_files = self.get_modified_files
         self.repo_handler.get_all_labels = self.get_all_labels
 
     # Patched methods:
@@ -57,6 +55,6 @@ class TestAutolabel:
         assert 'Docs' in self.pr_handler.labels
 
         # now check that all the subpackage labels are added:
-        assert 'io.fits' in self.pr_handler.labels
-        assert 'coordinates' in self.pr_handler.labels
-        assert 'visualization.wcsaxes' in self.pr_handler.labels
+        assert 'io.fits' in self.labels
+        assert 'coordinates' in self.labels
+        assert 'visualization.wcsaxes' in self.labels

--- a/astropy_bot/tests/test_autolabel.py
+++ b/astropy_bot/tests/test_autolabel.py
@@ -4,27 +4,21 @@
 
 from unittest.mock import MagicMock
 
-from astropy_bot.autolabel import autolabel_subpackages
+from astropy_bot.autolabel import autolabel
 
 
-class TestAutolabel:
+class AutolabelBase:
+
+    all_labels = ['analytic_functions', 'config', 'constants',
+                  'convolution', 'coordinates', 'cosmology',
+                  'io.ascii', 'io.fits', 'io.misc', 'io.misc.asdf',
+                  'io.registry', 'io.votable', 'modeling', 'nddata',
+                  'samp', 'stats', 'table', 'time', 'timeseries',
+                  'uncertainty', 'units', 'utils', 'visualization',
+                  'visualization.wcsaxes', 'vo', 'vo.conesearch',
+                  'wcs', 'wcs.wcsapi']
 
     def setup_method(self, method):
-
-        self.files = ['CHANGES.rst', 'astropy/io/fits/convenience.py',
-                      'astropy/io/fits/fitstime.py',
-                      'astropy/io/fits/tests/test_fitstime.py',
-                      'astropy/coordinates/baseframe.py',
-                      'astropy/visualization/wcsaxes/coordinate_helpers.py']
-
-        self.all_labels = ['analytic_functions', 'config', 'constants',
-                           'convolution', 'coordinates', 'cosmology',
-                           'io.ascii', 'io.fits', 'io.misc', 'io.misc.asdf',
-                           'io.registry', 'io.votable', 'modeling', 'nddata',
-                           'samp', 'stats', 'table', 'time', 'timeseries',
-                           'uncertainty', 'units', 'utils', 'visualization',
-                           'visualization.wcsaxes', 'vo', 'vo.conesearch',
-                           'wcs', 'wcs.wcsapi']
 
         self.pr_handler = MagicMock()
         self.pr_handler.number = 1234
@@ -46,10 +40,17 @@ class TestAutolabel:
     def set_labels(self, labels):
         self.labels = labels
 
-    # Tests:
+
+class TestAutolabel1(AutolabelBase):
+    files = ['CHANGES.rst', 'astropy/io/fits/convenience.py',
+             'astropy/io/fits/fitstime.py',
+             'astropy/io/fits/tests/test_fitstime.py',
+             'astropy/coordinates/baseframe.py',
+             'astropy/visualization/wcsaxes/coordinate_helpers.py']
+
     def test_autolabel_subpackages(self, app):
         with app.app_context():
-            autolabel_subpackages(self.pr_handler, self.repo_handler)
+            autolabel(self.pr_handler, self.repo_handler)
 
         # make sure 'Docs' stays in there:
         assert 'Docs' in self.pr_handler.labels
@@ -58,3 +59,19 @@ class TestAutolabel:
         assert 'io.fits' in self.labels
         assert 'coordinates' in self.labels
         assert 'visualization.wcsaxes' in self.labels
+
+
+class TestAutolabel2(AutolabelBase):
+    files = ['astropy/io/fits/tests/test_thing.py',
+             'docs/units/thing.rst']
+
+    def test_autolabel_subpackages(self, app):
+        with app.app_context():
+            autolabel(self.pr_handler, self.repo_handler)
+
+        # make sure 'Docs' stays in there:
+        assert 'Docs' in self.pr_handler.labels
+
+        # now check that all the subpackage labels are added:
+        assert 'io.fits' in self.labels
+        assert 'units' in self.labels

